subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -83,9 +73,7 @@ public void setDestinationDir(File destinationDir) {
     @SkipWhenEmpty
     @IgnoreEmptyDirectories
     @PathSensitive(PathSensitivity.NONE)
-    public FileCollection getTestResultDirs() {
-        return resultDirs;
-    }
+    public abstract ConfigurableFileCollection getTestResultDirs();
 
     private void addTo(Object result, ConfigurableFileCollection dirs) {
         if (result instanceof Test) {
```

Comment:
We could special case `result instanceof FileCollection` here so that we don't have to change anything else, but I think bringing this task up to modern standards is a better approach.

Reply:
I chose to implement new properties and add `@ReplacedBy` on the legacy accessors.